src/main/java/mcjty/rftools/items/netmonitor/GuiNetworkMonitor.java

```diff
@@ -196,8 +196,14 @@ private void populateList() {
                 String displayName = BlockInfo.getReadableName(block, coordinate, meta, mc.theWorld);
 
                 if (filter != null) {
-                    if (!displayName.toLowerCase().contains(filter)) {
-                        continue;
+                    try {
+                        if (!Pattern.matches(filter, displayName.toLowerCase())) {
+                            continue;
+                        }
+                    } catch (Exception e) {
```

Comment:
I expect that this should be catching `PatternSyntaxException` rather than Exception.

It may also make sense to cache the compiled regex as an instance field so that you don't end up paying the compile costs every time this method is called.

At the very least you should `Pattern.compile` outside of the loop, as it stands it will be compiled at every iteration